File: visitor/kotlin/Book.kt
package Visitor

// Temel sınıftır.
// Yeni davranışları Accept metotu ile alır.
// UML diyagramındaki Element yapısına denk gelmektedir.
internal abstract class Book {
    var name: String? = null
    var author: String? = null
    var categoryName: String? = null

    abstract fun accept(visitor: IBookExportVisitor)
}

File: visitor/typescript/book.ts
import {IBookExportVisitor} from "./book-export-visitor.interface";

// Temel sınıftır.
// Yeni davranışları Accept metotu ile alır.
// UML diyagramındaki Element yapısına denk gelmektedir.
export abstract class Book {
  public name: string;
  public author: string;
  public categoryName: string;

  public abstract accept(visitor: IBookExportVisitor): void;
}
